eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/LightningMessageTypes.scala

```diff
@@ -435,6 +439,30 @@ case class GossipTimestampFilter(chainHash: ByteVector32, firstTimestamp: Timest
 
 case class OnionMessage(blindingKey: PublicKey, onionRoutingPacket: OnionRoutingPacket, tlvStream: TlvStream[OnionMessageTlv] = TlvStream.empty) extends LightningMessage
 
+sealed trait PeerSwapMessage extends LightningMessage
+
+sealed abstract class JSonBlobMessage() extends PeerSwapMessage {
+  def json: String = {
+    Serialization.write(this)(PeerSwapJsonSerializers.formats)
+  }
+}
+
+case class SwapInRequest(protocolVersion: Long, swapId: String, asset: String, network: String, scid: String, amount: Long, pubkey: String) extends JSonBlobMessage with HasSwapId
```

Comment:
Should we use more restrictive types here, `ByteVector32` for swap ids for example ?

Reply:
Until the spec is changed from json to byte encodings our codec will have to work with swapId as a string. I couldn't figure out a way to parse the json into messages with fields of a different type.